### DISCUSSION TITLE
Linux Repositories

### DISCUSSION THREAD

**Discussion by gj82854** [BOINC/boinc#5315] [Q&A]
Are the repositories at /boinc.berkeley.edu/dl/linux usable and are they going to be updated on any sort of a regular basis? Is the strategy going forward to continue to rely on third parties to compile the latest code and maintain their own repositories for the different Linux distributions? As we have seen, this isn't always the best approach. How would I get the 7.22.x version of the client on Fedora? This is somewhat related to issue #4930 that is still open.

**Comment by AenBleidd** [comment]
No, they are not usable yet. We haven't tested them at all. We are planning to start alpha test of this approach with the next release.
But please keep in mind that this is not promised to work from the first try, so it might take several bug-fix releases to make it usable.

No, will not rely on the distro maintainers anymore. You should still see there release of BOINC, but the recommended way of installing it will be the official BOINC website.

Currently, you have no way to install 7.22.2 release of BOINC for Linux using **this** approach.
The first release that we will test using this approach is future 7.24.0.

**Comment by gj82854** [comment]
Understood, Thank You

**Comment by gj82854** [comment]
Vitalii,

Just as a test, I tried using the fc38 directory in the alpha repository fully understanding it might not work. Only encountered 2 things that needed to be fixed for the Fedora install.

(1). Install changed the ownership of the /var/lib/boinc directory to root from boinc. This resulted in the client not being able to write to the existing data directory.

(2). The systemd boinc-client.services file in /var/lib/systemd/system tries to start the boinc-client out of /usr/local/bin instead /usr/bin.

Once I fixed those 2 issues and entered systemctl start boinc-client, the 7.24 client started and is currently running a few test tasks just to see if it can download, execute, and return work. This was tested on a test box so that if anything went wrong it could wiped and retested if necessary.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by AenBleidd** [comment]
@gj82854, thank you for the testing and a very valuable input.
If you want to continue testing it, please feel free to create any new issues in the issue tracker here: https://github.com/BOINC/boinc/issues/new?assignees=&labels=&projects=&template=bug_report.md

I have created two issues based on your testing:
- #5324
- #5325

**Comment by computezrmle** [comment]
Shouldn't BOINC follow the FHS standard published here?
https://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html

If so, the installation directory may depend on who offers the package.
Distribution maintainers should use /usr/bin while self compiled packages or packages not provided by distribution maintainers should go to /usr/local/bin.

This ensures a local installation is not overwritten/removed by a distribution upgrade.

It should also be checked if the path is correctly set since content and order are important.
It should look a bit like this:
PATH=<user_specific:>/usr/local/bin:/usr/bin:/bin<:may_be_more>


Same applies to BOINC's systemd service file.
Provided by the distribution it should be in /usr/lib/systemd/system, else it should be in /etc/systemd/system.

Since systemd service files must include the full path to an executable they consequently need to look either like this (maintained by a distribution):
ExecStart=/usr/bin/boinc

or like this (not maintained by a distribution):
ExecStart=/usr/local/bin/boinc

**Reply by AenBleidd** [reply]
@computezrmle, thank you for the link. I'll read it carefully and fix our packages if needed.

**Comment by AenBleidd** [comment]
This is done and released: https://boinc.berkeley.edu/linux_install.php
